packages/trpc/server/routers/viewer/eventTypes/update.handler.ts

```diff
@@ -241,13 +241,13 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
     }
   }
 
-  if (input?.price || input.metadata?.apps?.stripe?.price) {
-    data.price = input.price || input.metadata?.apps?.stripe?.price;
+  if (input.metadata?.apps?.stripe?.price) {
```

Comment:
Also instead of multiple conditionals, can we abstract this into a function? getPaymentCredential from metadata or something like that.

Reply:
Hi! I'm currently working in adding `MercadoPago` payment integration to Cal.com (work in progress [here](https://github.com/ubinatus/cal.com/tree/feature/mercadopago). Should create a PR later today). 

Regarding @zomars  comment, I've thought about the following:

```ts
  // TODO: This could be a function to abstract the payment setters in the eventType as only one is allowed.
  const paymentAppIds = ["stripe", "paypal", "mercadopago"];
  if (input?.price) {
    const paymentCredentials = await ctx.prisma.credential.findFirst({
      where: {
        userId: ctx.user.id,
        appId: {
          in: paymentAppIds,
        },
      },
      select: {
        type: true,
        key: true,
      },
    });

    paymentAppIds.forEach((paymentAppId) => {
      if (
        paymentCredentials.appId === paymentAppId &&
        input.metadata?.apps &&
        input.metadata?.apps[paymentAppId] &&
        input.metadata?.apps[paymentAppId].enabled
      ) {
        data.price = input.price || input.metadata?.apps[paymentAppId].price;
        if (paymentAppId === "stripe") {
          const { default_currency } = stripeDataSchema.parse(paymentCredential.key);
          data.currency = default_currency;
        } else {
          data.currency = input.metadata?.apps[paymentAppId].currency.toLowerCase();
        }
      }
    });
  }
```

That should only set the price/currency from the enabled payment app, since the user could now have more than 1 payment `Credential` but not necessarily the first found is the one that belongs to the `EventType`.

Best,
JA

Reply:
Made some cleaning, here's the updated code:

```ts
  const paymentAppIds = ["stripe", "paypal", "mercadopago"];
  if (input.metadata?.apps) {
    const paymentCredentials = await ctx.prisma.credential.findMany({
      where: {
        userId: ctx.user.id,
        appId: {
          in: paymentAppIds,
        },
      },
      select: {
        type: true,
        appId: true,
        key: true,
      },
    });

    paymentCredentials.forEach((paymentCredential) => {
      const appId = paymentCredential.appId! as keyof typeof input.metadata.apps;
      const appMetadata = input.metadata?.apps ? input.metadata?.apps[appId] : null;
      if (appMetadata?.enabled) {
        data.price = input.price || appMetadata.price;
        if (appId === "stripe") {
          const { default_currency } = stripeDataSchema.parse(paymentCredential.key);
          data.currency = default_currency;
        } else {
          data.currency = appMetadata.currency.toLowerCase();
        }
      }
    });
  }
  ```